stdlib/public/core/ShadowProtocols.swift

```diff
@@ -182,10 +182,36 @@ public protocol _NSNumber {
   var objCType: UnsafePointer<Int8> { get }
 }
 
+/// A shadow for the API of NSString we will use in the core stdlib.
+@objc
+public protocol _NSStringCore {
+  func length() -> Int
+  func characterAtIndex(_ index: Int) -> UInt16
+
+  func getCharacters(
+    _ buffer: UnsafeMutablePointer<UInt16>,
+    range aRange: _SwiftNSRange)
+
+  // SPI APIs
+  func _fastCharacterContents() -> UnsafeMutablePointer<UInt16>?
+}
+
+// See _NSSet above for why this exists
+@unsafe_no_objc_tagged_pointer @objc
+public protocol _NSString : _NSStringCore {
+  func getCharacters(_ buffer: UnsafeMutablePointer<UInt16>)
+
+  // SPI APIs
+  // TODO: undo Uint8 being used as a proxy for bool
+  func _fastCStringContents(_ nullTerminationRequired: UInt8) 
```

Comment:
Summarizing: this is a bad hack that probably works fine in practice for the time being. When https://reviews.llvm.org/D26234 is merged, we can use that to do the *strictly correct* thing and provide a `#define` in SwiftShims.